@ecomplus/storefront-components/src/js/TheProduct.js

```diff
@@ -395,7 +395,10 @@ export default {
       const { customizations } = this
       this.$emit('buy', { product, variationId, customizations })
```

Comment:
Como usa o `customizations` no evento _buy_ também, acho que é melhor já clonar o array aqui em vez de só no `ecomCart.addProduct`, pra evitar o mesmo erro caso `!this.canAddToCart` e o `addProduct` seja manipulado no parente...

Reply:
Faz sentido